gelf/reader.go

```diff
@@ -64,67 +75,51 @@ func (r *Reader) Read(p []byte) (int, error) {
 func (r *Reader) ReadMessage() (*Message, error) {
 	cBuf := make([]byte, ChunkSize)
 	var (
-		err        error
-		n, length  int
-		cid, ocid  []byte
-		seq, total uint8
-		cHead      []byte
-		cReader    io.Reader
-		chunks     [][]byte
+		err       error
+		n         int
+		chunkHead []byte
+		cReader   io.Reader
 	)
+	var message []byte
 
-	for got := 0; got < 128 && (total == 0 || got < int(total)); got++ {
+	for {
 		if n, err = r.conn.Read(cBuf); err != nil {
 			return nil, fmt.Errorf("Read: %s", err)
 		}
-		cHead, cBuf = cBuf[:2], cBuf[:n]
-
-		if bytes.Equal(cHead, magicChunked) {
-			// fmt.Printf("chunked %v\n", cBuf[:14])
-			cid, seq, total = cBuf[2:2+8], cBuf[2+8], cBuf[2+8+1]
-			if ocid != nil && !bytes.Equal(cid, ocid) {
-				return nil, fmt.Errorf("out-of-band message %v (awaited %v)", cid, ocid)
-			} else if ocid == nil {
-				ocid = cid
-				chunks = make([][]byte, total)
+		chunkHead, cBuf = cBuf[:2], cBuf[:n]
+		if bytes.Equal(chunkHead, magicChunked) {
```

Comment:
You should probably check the length of cBuf, to ensure it is long enough to have a full messageID

Reply:
done